Teleport Join Tests ✨

EvolveArt opened the pull request.

EvolveArt committed:
feat: add mocks & test init (wip) ✨

EvolveArt committed:
fix: code review fixes 🔨

EvolveArt committed:
feat: file functions + new tests ✨

EvolveArt committed:
feat: more tests + cure function ✨

EvolveArt committed:
feat: more tests + join fixes ✨

EvolveArt committed:
feat: last tests ✨

EvolveArt committed:
Merge branch 'domain' into evolve/teleport-join

EvolveArt committed:
fix: fixing tests 🔨

EvolveArt committed:
fix: more tests fixes 🔨

EvolveArt committed:
fix: more test fixes 🔨

- added `mock_vat`
- update hardhat plugin

EvolveArt committed:
fix: more test fixes 🔨

- remove bytes32 conversion
- temp remove `daiSent` & `totalFees` checks

EvolveArt committed:
fix: more test fixes 🔨

- write dai address at initialization

EvolveArt committed:
fix: more test fixes 🔨

EvolveArt committed:
fix: minor test fixes 🔨

- add signed/signed sub function (from warp)
- fix bug in `vat_mock`

EvolveArt committed:
fix: teleport join tests ✨

- todo: handle negative integers in js

EvolveArt committed:
fix: ci ♻️

EvolveArt committed:
fix: ci ♻️

EvolveArt committed:
fix: ci ♻️

EvolveArt committed:
fix: ci ♻️

EvolveArt committed:
fix: ci ♻️

EvolveArt committed:
fix: ci ♻️

maciejka commented on contracts/starknet/safe_math.cairo:
```diff
@@ -168,3 +169,41 @@ func div{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
     return (c,);
 }
+
+func sub_signed256{range_check_ptr, bitwise_ptr: BitwiseBuiltin*}(lhs: Uint256, rhs: Uint256) -> (
```
input types should be `Int256`, right?

maciejka commented on contracts/starknet/safe_math.cairo:
```diff
@@ -168,3 +169,41 @@ func div{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    return (res=res);
+}
+
+func add_signed256{range_check_ptr, bitwise_ptr: BitwiseBuiltin*}(lhs: Uint256, rhs: Uint256) -> (
```
Same as above: `Int256` not `Uint256`

maciejka commented on contracts/starknet/vat.cairo:
```diff
@@ -720,8 +719,8 @@ func frob{
         let (ilk_debt) = mul(Art, ilk.rate);
-        let (line_ok) = le(ilk_debt, ilk.line);
-        let (Line_ok) = le(debt, ilk.line);
+        let (line_ok) = uint256_le(ilk_debt, ilk.line);
```
Idea behind `le` was that it is easier to read. Especially when function results will be usable in expressions, having shorter expressions will be important. Although at that point simple `<` might be supported as well.

EvolveArt replied:
I don't know if you've seen but there is a discussion on-going regarding felt assertions. https://github.com/OpenZeppelin/cairo-contracts/issues/433

EvolveArt committed:
fix: type name ✍️

EvolveArt committed:
feat: solidity comments ✍️

EvolveArt committed:
feat: teleport join changes ✨

- update code to follow https://github.com/makerdao/dss-teleport/commits/routing-v2/src/TeleportJoin.sol

maciejka commented on contracts/starknet/teleport_join.cairo:
```diff
@@ -277,25 +276,100 @@ func Settle(source_domain: felt, amount: Uint256) {
     VatLike.hope(vat_, daiJoin_);
     let (dai) = DaiJoinLike.dai(daiJoin_);
+    _dai.write(dai);
     TokenLike.approve(dai, daiJoin_, Uint256(2 ** 128 - 1, 2 ** 128 - 1));
```
`2 ** 128 - 1` should be a constant

maciejka commented on contracts/starknet/teleport_join.cairo:
```diff
@@ -379,13 +477,62 @@ func deny{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(user:
+//     * @param teleportGUID Struct which contains the whole teleport data
+//     * @param hashGUID Hash of the prev struct
+//     **/
+func _register{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
```
Solidity declaration is missing

maciejka commented on test/teleport_join.ts:
```diff
@@ -0,0 +1,1914 @@
+import { expect } from 'chai';
```
In general original solidity test names are missing.

EvolveArt committed:
fix: remove python test files 🔨

EvolveArt committed:
fix: review changes ✍️

EvolveArt committed:
fix: test solidity names ✍️

EvolveArt committed:
fix: switch to docker env ♻️
